### ISSUE TITLE
[Bug] fatal: remote error: upload-pack: not our ref

### ISSUE THREAD

**Issue by canivel** [mit-dci/terraform-aws-opencbdc-tctl#27]
Hi, ECS task test-controller is failing with the error:
Fetching submodule 3rdparty/secp256k1
fatal: remote error: upload-pack: not our ref
fatal: the remote end hung up unexpectedly
panic: runtime error: index out of range [-1]]

![image](https://user-images.githubusercontent.com/201984/196266563-33924570-f82b-4db4-971c-6f20dc8047e1.png)

**Comment by HalosGhost** [comment]
@canivel thank you for reporting! A lot of us are surprisingly busy this week, so you might not get a response/resolution right away, but know that this is on our radar!

**Comment by canivel** [comment]
any news?

**Comment by HalosGhost** [comment]
Unfortunately “this week” turned into “this month”, but this needs to be addressed. I'll try to take a look at this personally in the next day or so, and I'm also going to chat with a few people more knowledgeable than I on this subject. ☺

**Comment by bvkin** [comment]
Hello @canivel, is your environment using the mit-dci version of the test controller, i.e., [this](https://github.com/mit-dci/opencbdc-tctl) one?

**Comment by HalosGhost** [comment]
(by the way, @bvkin is definitely one of those more-knowledgeable people!)

**Comment by canivel** [comment]
Hi bvkin I will check the version and revert back, but as far as I understand probably is using the last version, I did not changed anything on the deployment

**Comment by canivel** [comment]
@bvkin from the variables.tf
![image](https://user-images.githubusercontent.com/201984/199542185-6e1b3dcf-6cc4-4b62-bca1-f2a49ba1b14b.png)
i just tried again from scratch and same error:
![image](https://user-images.githubusercontent.com/201984/199542734-f67f0491-67fe-4fa9-9e64-41841e63f6a6.png)
The error happen on the last stage of the Code pipeline : Deploy > Controller

**Comment by canivel** [comment]
hello? any news?

**Comment by HalosGhost** [comment]
Yes! We're working to reproduce this bug in our setup (we're standing up a full copy of the environment from-scratch).

**Comment by canivel** [comment]
Sorry to ask again but any luck?

**Comment by HalosGhost** [comment]
No worries and no problem. The short version, I believe, is that the test controller is a bit out-of-date compared to the transaction-processor and the terraform repo. I have a “release candidate” branch setup on the test controller which includes a lot of fixes/features/improvements that will hopefully solve most issues. I'd wonder if you can retarget your `test_controller_github_repo_branch` to point to `rc/2022-Nov-14`. Please let me know if that solves the issue for you or if it's still a problem. That double-check and testing would be incredibly valuable!

Note that the reason those changes haven't been pushed to `trunk` quite yet is because I'm still testing them to ensure everything works, so running into other issues may be possible. ☺

**Comment by canivel** [comment]
Awesome it's all running now, will test the components and let you know.
The issue with the requirements cp to S3 still there, I commented the line 533 of main./tf and it compiled fine
- aws --endpoint-url $S3_INTERFACE_ENDPOINT s3 cp ./requirements.txt s3://$S3_BUCKET/test-controller-agent/
Thank you!

`

[Container] 2022/11/15 18:20:12 Running command aws --endpoint-url $S3_INTERFACE_ENDPOINT s3 cp ./requirements.txt s3://$S3_BUCKET/test-controller-agent/
--
108 |  
109 | The user-provided path ./requirements.txt does not exist.
110 |  
111 | [Container] 2022/11/15 18:20:12 Command did not exit successfully aws --endpoint-url $S3_INTERFACE_ENDPOINT s3 cp ./requirements.txt s3://$S3_BUCKET/test-controller-agent/ exit status 255
112 | [Container] 2022/11/15 18:20:12 Phase complete: BUILD State: FAILED
113 | [Container] 2022/11/15 18:20:12 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: aws --endpoint-url $S3_INTERFACE_ENDPOINT s3 cp ./requirements.txt s3://$S3_BUCKET/test-controller-agent/. Reason: exit status 255

`

**Comment by HalosGhost** [comment]
I've still not been able to reproduce this issue. You said you commented out `main.tf:533`? Can you link to the line? The place I'm looking doesn't seem to map to what you're referring to.

**Comment by canivel** [comment]
![image](https://user-images.githubusercontent.com/201984/203377843-1149d6ab-c5d5-4710-bf89-ca3cb392da89.png)

**Comment by HalosGhost** [comment]
clickable perma-link: https://github.com/mit-dci/terraform-aws-opencbdc-tctl/blob/9bf8815aec22d487d862bf4c51aa49fa1f5eef17/modules/test-controller-deploy/main.tf#L534

**Comment by HalosGhost** [comment]
@bvkin we didn't run into this issue during our testing; any idea what might have enabled us to get by unscathed?
